src/Faithlife.Data/DbConnectorCommand.cs

```diff
@@ -345,18 +342,68 @@ string Replacement(Match match)
 					commandText = Regex.Replace(commandText, $@"([?@:]{Regex.Escape(name)})\.\.\.",
 						Replacement, RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
 
-					// if special syntax wasn't found, just add the parameter, for databases that support collections directly
-					if (itemCount == -1)
-						AddCommandParameter(name, value);
+					// if special syntax wasn't found, leave the parameter alone, for databases that support collections directly
+					if (itemCount != -1)
+					{
+						parameters = DbParameters.Create(parameters.Take(index).Concat(replacements).Concat(parameters.Skip(index + 1)));
+						index += replacements.Count;
+					}
+					else
+					{
+						index += 1;
+					}
 				}
 				else
 				{
-					AddCommandParameter(name, value);
+					index += 1;
 				}
 			}
 
-			command.CommandText = commandText;
+			IDbCommand command;
+			var transaction = Connector.Transaction;
+
+			var preparedCache = IsPrepared ? Connector.PreparedCommandCache : null;
+			if (preparedCache != null)
+			{
+				if (preparedCache.TryGetValue(commandText, out command))
+				{
+					command.Parameters.Clear();
```

Comment:
Reusing the parameters would be preferable from a GC and efficiency perspective... but it may be negligible in practice.

Reply:
Yeah, I'm willing to introduce that optimization later if needed. This is simpler and more flexible.